doc/nrf/libraries/modem/at_shell.rst

```diff
@@ -0,0 +1,56 @@
+.. _lib_at_shell:
+
+AT shell
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can use this library to add an AT shell command to a shell.
+This enables you to send AT commands to the modem and receive responses.
+When the library is enabled, the shell will have a command ``at``.
+
+You can also enable AT command notifications.
+
+* Enable AT command notifications:
+
+  .. code-block:: console
+
+     at events_enable
+
+* Disable AT command notifications:
+
+  .. code-block:: console
+
+     at events_disable
+
+
+Configuration
+*************
+
+The library expects that you have enabled a shell.
+To enable and configure the shell, use the following Kconfig options:
+
+* :kconfig:`CONFIG_AT_SHELL` - Enables the AT Host library
+* :kconfig:`CONFIG_AT_SHELL_CMD_RESPONSE_MAX_LEN` - Maximum AT command response length
+
+To send AT read commands (ending with ``?``), disable the shell wildcards using the following Kconfig option:
+
+* :kconfig:`CONFIG_SHELL_WILDCARD`
+
+Dependencies
+************
+
+This library uses the following |NCS| libraries:
+
+* :ref:`at_monitor_readme`
+
+This library uses the following `sdk-nrfxlib`_ libraries:
+
+* :ref:`nrfxlib:nrf_modem_at`
+
+
```

Comment:
```suggestion
API documentation
*****************

| Header file: :file:`path_to_the_header_file`
| Source file: :file:`lib/at_shell/at_shell.c`
```
Add the header file and the path to it.

Reply:
The library actually doesn't have a header file and no public API, only Kconfig options, so I removed this section

Reply:
Should we mention this, that is, the fact there is no header file. As a developer not familiar with shell and AT shell, I'd be looking for it. And not mentioning it in documentation can mean it doesn't exist, or it's been forgotten. So to rule this out. I'd clearly say somewhere that this library doesn't have an API or header file but only configurations.
I don't know if we've said this in some other library somewhere.

@peknis What do you think?

Reply:
A good idea. We have an example of such case in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/modem/at_host.html.

Reply:
Alright, I'll add a similar section to that, thanks for the link @peknis 
Btw, there might be a header file coming down the road to allow for custom print/format functions